packages/extension/src/libs/RPChProvider/index.ts

```diff
@@ -0,0 +1,65 @@
+import RPChSDK, { type Ops } from "@rpch/sdk";
+import { AbstractProvider } from "web3-eth/node_modules/web3-core"; // I had to import from web3-eth dependencies as web3-core has other version
+import { Result, Error as sdkError } from "@rpch/sdk/build/jrpc";
+
+export const getSupportedRpchProvider = (
+  rpcUrl: string
+): string | RPChProvider => {
+  if (/^ws(s)?:\/\//i.test(rpcUrl)) {
+    return rpcUrl;
+  }
+  return new RPChProvider(rpcUrl);
+};
+
+export class RPChProvider implements AbstractProvider {
+  sdk: RPChSDK;
+
+  constructor(rpcUrl: string) {
+    const ops: Ops = {
+      discoveryPlatformEndpoint:
+        process.env.VUE_APP_DISCOVERY_PLATFORM_API_ENDPOINT || undefined,
+      provider: rpcUrl,
+    };
+
+    // TODO: Remove after confirmation and testing
+    console.log("RPCh: CREATING SDK INSTANCE with OPS ", ops);
+    console.log("RPCh: Client ID ", process.env.VUE_APP_RPCH_SECRET_TOKEN);
+
+    if (!process.env.VUE_APP_RPCH_SECRET_TOKEN) {
+      throw new Error("MISSING RPCH SECRET TOKEN");
+    }
+
+    this.sdk = new RPChSDK(process.env.VUE_APP_RPCH_SECRET_TOKEN, ops);
+  }
+
+  private isError(jsonRes: Result | sdkError): jsonRes is sdkError {
+    return "error" in jsonRes;
+  }
+
+  sendAsync(
+    payload: Parameters<AbstractProvider["sendAsync"]>[0],
+    callback: Parameters<AbstractProvider["sendAsync"]>[1]
+  ) {
+    this.sdk
+      .send({
+        ...payload,
+        jsonrpc: "2.0",
+      })
+      .then(async (res) => {
+        const jsonRes: Result | sdkError = await res.json();
+        if (this.isError(jsonRes)) {
+          callback?.({ name: "", ...jsonRes.error });
```

Comment:
This is not an error. A valid response can have "error" as an attribtue

Reply:
Done.